src/Microsoft.Health.Fhir.Core/Features/Search/SearchIndexEntry.cs

```diff
@@ -16,21 +17,21 @@ public class SearchIndexEntry
         /// <summary>
         /// Initializes a new instance of the <see cref="SearchIndexEntry"/> class.
         /// </summary>
-        /// <param name="paramName">The search parameter name.</param>
+        /// <param name="searchParameter">The search parameter</param>
         /// <param name="value">The searchable value.</param>
-        public SearchIndexEntry(string paramName, ISearchValue value)
+        public SearchIndexEntry(SearchParameter searchParameter, ISearchValue value)
         {
-            EnsureArg.IsNotNullOrWhiteSpace(paramName, nameof(paramName));
+            EnsureArg.IsNotNull(searchParameter, nameof(searchParameter));
             EnsureArg.IsNotNull(value, nameof(value));
 
-            ParamName = paramName;
+            SearchParameter = searchParameter;
             Value = value;
         }
 
         /// <summary>
-        /// Gets the parameter name.
+        /// Gets the search parameter
         /// </summary>
-        public string ParamName { get; }
+        public SearchParameter SearchParameter { get; }
```

Comment:
What is this change for? It seems like the change in `CodeableConceptToSearchValueTypeConverter` to avoid duplicate doesn't actually require this change and this chagne is not being used anywhere?

Reply:
It's not required. It's just a refactoring in preparation for the SQL provider